src/mechanics/interpolation/InterpolationTriangleQuadratic.h

```diff
@@ -0,0 +1,49 @@
+#pragma once
+#include <eigen3/Eigen/Core>
+#include "mechanics/interpolation/InterpolationSimple.h"
+#include "mechanics/elements/ElementShapeFunctions.h"
+
+namespace NuTo
+{
+class InterpolationTriangleQuadratic : public InterpolationSimple
+{
+public:
+    InterpolationTriangleQuadratic(int dofDimension)
+        : mDofDimension(dofDimension)
+    {
+    }
+
+    std::unique_ptr<InterpolationSimple> Clone() const override
+    {
+        return std::make_unique<InterpolationTriangleQuadratic>(*this);
+    }
+
+    ShapeFunctions GetShapeFunctions(const NaturalCoords& naturalIpCoords) const override
+    {
+        return ShapeFunctions2D::ShapeFunctionsTriangleOrder2(naturalIpCoords);
```

Comment:
Can we start moving this code here, instead of having this huge `ElementShapeFunctions.cpp`?

Reply:
I left it as it is since I was not sure how it should look like in an ideal world. And wanted to save some work. 

Maybe
~~~cpp
Triangle<TOrder>
Quad<TOrder, TType>
// with TType being on of
struct Equidistant{};
struct Lobatto{};
struct Serendipity{};
~~~
or "simply"
~~~cpp
Interpolation<TShape, TOrder, TType>
// with a corresponding
Integration<TShape, TOrder, TType>
// where TShape has to match
~~~
(maybe see #34)

Who knows? Not me. (We never lost control... :) )